spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/jmx/EndpointMBean.java

```diff
@@ -90,7 +98,13 @@ public Object invoke(String actionName, Object[] params, String[] signature)
 					+ "' has no operation named " + actionName;
 			throw new ReflectionException(new IllegalArgumentException(message), message);
 		}
-		return invoke(operation, params);
+		ClassLoader previousClassLoader = ClassUtils.overrideThreadContextClassLoader(this.classLoader);
```

Comment:
This may throw a `SecurityException`. I think we should set the TCCL on a best-effort basis so any `SecurityException` that's thrown should be caught (and logged) and we should then proceed with the `invoke`.